File: LeetCode/py/1768.py
#Link: https://leetcode.com/problems/merge-strings-alternately/

class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        i = 0
        result = ''
        while i < len(word1) and i < len(word2):
            result += word1[i]
            result += word2[i]
            i += 1
        while i < len(word1):
            result += word1[i]
            i += 1
        while i < len(word2):
            result += word2[i]
            i += 1
        return result
        

File: LeetCode/cpp/1768.cpp
//Link: https://leetcode.com/problems/merge-strings-alternately/

class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string result;
        int i = 0;
        while(i < word1.size() && i < word2.size()) {
            result += word1[i];
            result += word2[i];
            i++;
        }
        while(i < word1.size()) {
            result += word1[i];
            i++;
        }
        while(i < word2.size()) {
            result += word2[i];
            i++;
        }
        return result;
    }
};
